core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -254,8 +254,16 @@ object SparkEnv extends Logging {
           if (port == 0 || rpcEnv.address == null) {
             port
           } else {
-            rpcEnv.address.port + 1
+
+            if(Utils.hasPortRange(conf)){
+              0 // pick one form the range
```

Comment:
At this point at the code, normally we would pick rpcEnv.address.port + 1 to create the next actorsytem. But since we dont know if that falls within range and we dont want to do this check at this point we pass zero so at the lowest level this is checked and netty does the retries. We dont want this upper ligher to pick a port and try it there we want this logic to be transparent and only pass if this is going to be a specific port or a random any port from the range. There is no other option to pass there to triggere the correct code path, zero by convention means any random free port (not jsut in java api but is an OS feature also used by spark in netty classes).